### PR TITLE
Add note

### PR DESCRIPTION
**Pull request by AnkurChunekar**
- Add rich text editor.
- Add new note feature.
- Add pin notes feature
- [Preview](https://deploy-preview-3--velnotes-react.netlify.app/)

> Single note page will be added later. Haven't implemented Label, Color and priority features yet.

### CODE REVIEW

File: src/components/modals/CreateNoteModal.jsx [hunk 1]
```diff
@@ -1,6 +1,39 @@
...
+
+  const initialInputData = {
+    title: "",
+    content: "<p><br></p>",
+    isPinned: false,
+    color: "green",
+    tags: [],
+    priority: "low",
+    date: currentDate.toDateString().slice(4),
+    time: currentDate.getHours() + ":" + currentDate.getMinutes(),
```
Comment (kart-c) [inline src/components/modals/CreateNoteModal.jsx]: The ~~date~~ time is formatting to `22:1` when the time was `22:01`. Can you check into this? Also there is an alternative to this. The backend folder has a `formatDate` function in authUtils. You can use that to show current date and time instead.

File: src/context/notes-context.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        headers: { authorization: token },
+      });
+      console.log(response);
+      // if (response.status === 201) {
+      //   // notesDispatch({
+      //   //   type: "UPDATE_NOTES",
+      //   //   payload: { notesList: data.notes },
+      //   // });
+      //   toast.success("Note added successfully.");
+      // }
```
Comment (kart-c) [inline src/context/notes-context.js]: You can clean this up if not required

**Review by kart-c — APPROVED** [review]
The UI is really clean. Amazing job with the UI.

**Review by dipteekhd — APPROVED** [review]

File: src/context/notes-context.js (hunk 2 above)
Comment (AnkurChunekar) [reply src/context/notes-context.js]: Thanks for pointing this out

**Comment by vp5h** [comment]
- Can you add some default notes?
- Archive and trash pages redirects to blank pages and even the back button does not work there

**Review by vp5h — APPROVED** [review]

**Comment by AnkurChunekar** [comment]
> * Can you add some default notes?
> * Archive and trash pages redirects to blank pages and even the back button does not work there

Thankyou for pointing that out will look into it.

File: src/components/noteCard/NoteCard.jsx [hunk 3]
```diff
@@ -1,24 +1,22 @@
...
 
-export function NoteCard() {
+export function NoteCard({ noteData }) {
+  const { title, content, date, time } = noteData;
+
   return (
     <div className="note flex flex-column">
       <header className="flex ai-center jc-space-b">
-        <h2 className="title fw-600">My Notes</h2>
+        <h2 className="title fw-600"> {title} </h2>
```
Comment (dipteekhd) [inline src/components/noteCard/NoteCard.jsx:17]: (On text overflow...can show tooltip)
Here you can use title attribute on h2 tag  <h2 title={title}>

Comment (AnkurChunekar) [reply src/components/noteCard/NoteCard.jsx:17]: Thankyou for the feedback

File: src/components/modals/CreateNoteModal.jsx (hunk 1 above)
Comment (AnkurChunekar) [reply src/components/modals/CreateNoteModal.jsx]: Thanks for the suggestions.

**Pull request merged**